packages/core/test/predictableExec.test.ts

```diff
@@ -688,4 +688,39 @@ describe('predictableExec', () => {
       })
       .start();
   });
+
+  it('should be possible to send immediate events to initially invoked actors', () => {
+    const child = createMachine({
+      predictableActionArguments: true,
+      on: {
+        PING: {
+          actions: sendParent({ type: 'PONG' })
+        }
+      }
+    });
+
+    const machine = createMachine({
+      predictableActionArguments: true,
+      initial: 'waiting',
+      states: {
+        waiting: {
+          invoke: {
+            id: 'ponger',
+            src: child
+          },
+          entry: send({ type: 'PING' }, { to: 'ponger' }),
+          on: {
+            PONG: 'done'
+          }
+        },
+        done: {
+          type: 'final'
+        }
+      }
+    });
+
+    const service = interpret(machine).start();
+
+    expect(service.getSnapshot().value).toBe('done');
```

Comment:
In real-life scenarios, we can't assume that this is synchronous. Will the `machine` still get the `PONG` event _after_ it settles on the initial state (it should)? 

Maybe we can test this by instead listening for `.onDone(done)` so we don't depend on it being sync 🤔

Reply:
> In real-life scenarios, we can't assume that this is synchronous. Will the machine still get the PONG event after it settles on the initial state (it should)?

Yes, in this case, the test case is really more about the child receiving an event than about the parent receiving the response - I could slim it down.

> Maybe we can test this by instead listening for .onDone(done) so we don't depend on it being sync 🤔

I agree that those conceptually are async. However, I think that we also need to test synchronous logic as sync code often exposes more edge cases in scheduling etc. So it feels more beneficial to prefer sync tests in our test suite when possible - despite the mental model being async.